internal/util/k8s.go

```diff
@@ -180,5 +182,6 @@ func GenerateTagsForObject(obj client.Object) []*string {
 	if gvk.Version != "" {
 		tags = append(tags, K8sVersionTagPrefix+gvk.Version)
 	}
+	tags = append(tags, annotations.ExtractUserTags(obj.GetAnnotations())...)
```

Comment:
Do we need to process duplicate tags (like `konghq.com/tags:"a,a"`) or "reserved" rags (like `konghq.com/tags:"k8s-name:xxx"`)? 
I think we should run a `Unique` on tags, and omit the tags that are possible to confuse with generated tags to annotate k8s metadata, like `k8s-name:***`.

Reply:
Yeah, I agree that we could drop user-defined yet reserved tags by verifying that a tag doesn't begin with a "key" of one of those. That should probably also land in the feature documentation to make it clear what values are not allowed.

Reply:
Is there a built-in that handles this? I didn't think there was, and didn't think it was worth an extra dependency or code to handle it.

Duplicates will indeed break, but it falls into the odd area where you either:

- Know that we add the generated tags and added your own duplicates of them.
- Added duplicates of your own tags, where they're clearly visible (there's only one place you can set them).

Those both feel like the user needs to know enough about the system to understand how to break it, and then intentionally breaks it, rather than a footgun where you may reasonably break it by accident.

Reply:
I think the following would get rid of duplicates and make the ones added by us win over the user's ones:
```go
user := annotations.ExtractUserTags(obj.GetAnnotations())
tags = lo.Union(tags, user)
```

Reply:
Those libraries have the worst names :neutral_face: